### ISSUE TITLE
False positive "f-string" message

### ISSUE THREAD

**Issue by sam-s** [astral-sh/ruff#27578]
### Summary

ruff raises [f-string](https://docs.astral.sh/ruff/rules/f-string/) on *multiple* use of the same value, e.g.
```python
"use a value {v} several {v} times {v}".format(v=expensive_function())
```

while it is possible to rewrite the code to use f-strings, it would not be an obvious improvement.

### Version

ruff 0.16.2

**Comment by DresdenGman** [comment]
This looks like a plausible false positive with a small reproduction surface. I can take a look at whether the diagnostic should ignore repeated uses of the same value in a `.format(...)` call, or whether the rule needs a narrower trigger.

**Comment by MichaReiser** [comment]
This was added in https://github.com/astral-sh/ruff/issues/6217

I think this can easily be rewritten if the expression isn't deeply nested, e.g. 

```py
v=expensive_function()
f"use a value {v} several {v} times {v}"
```

The downside is that it introduces a local variable, but I think that's fine. 

Pulling out the variable in a more nested expression might be more difficult, but i'm not sure it justifies making an exception (might be a case where using a `ruff: ignore` is justified.)

**Comment by DresdenGman** [comment]
I think that direction is acceptable for the common case. Pulling out a temporary local variable is a reasonable rewrite when the expression is not deeply nested, and I agree that the deeper/nastier cases may be better handled with an explicit `ruff: ignore` rather than complicating the rule. If you prefer, I can try a minimal tweak that only narrows the false positive without broadening the rule surface.

**Comment by MichaReiser** [comment]
>  If you prefer, I can try a minimal tweak that only narrows the false positive without broadening the rule surface.

What's the false positive?

**Comment by DresdenGman** [comment]
The false positive is on a `.format(...)` call where the same value is used multiple times, for example:\n\n```py\n"use a value {v} several {v} times {v}".format(v=expensive_function())\n```\n\nRuff currently suggests converting this to an f-string, but the rewrite is only obviously better if we first pull out a temporary variable:\n\n```py\nv = expensive_function()\nf"use a value {v} several {v} times {v}"\n```\n\nSo the diagnostic is technically correct about the transformation, but misleading in the common case because it implies a direct swap is the right fix. For deeper expressions, I agree `ruff: ignore` may be the cleaner choice.

**Comment by MichaReiser** [comment]
I'm also not able to reproduce this. 

Repeated function calls are skipped https://github.com/astral-sh/ruff/commit/461cdad53a1569d6b6dfa242734b140bcad1b7b7

Unless you're using a very old Ruff version, in which case I strongly recommend you upgrading
